iphone/cli/commands/_build.js

```diff
@@ -5817,99 +5818,16 @@ iOSBuilder.prototype.copyResources = function copyResources(next) {
-										contents: originalContents,
-										symbols: []
-									};
-									this.cli.createHook('build.ios.compileJsFile', this, function (r, from, to, cb2) {
-										// Read the possibly modified file contents
-										const source = r.contents;
-										// Analyze Ti API usage, possibly also minify/transpile
-										// DO NOT TRANSPILE CODE inside SDK's common folder. It's already transpiled!
-										const transpile = from.startsWith(sdkCommonFolder) ? false : this.transpile;
-										const minify = from.startsWith(sdkCommonFolder) ? false : this.minifyJS;
-										const analyzeOptions = {
-											filename: from,
-											minify,
-											transpile,
-											sourceMap: this.sourceMaps || this.deployType === 'development',
-											resourcesDir: this.xcodeAppDir,
-											logger: this.logger,
-											targets: {
-												ios: this.minSupportedIosSdk
-											}
-										};
-
-										try {
-											const modified = jsanalyze.analyzeJs(source, analyzeOptions);
-											const newContents = modified.contents;
-
-											// we want to sort by the "to" filename so that we correctly handle file overwriting
-											this.tiSymbols[to] = modified.symbols;
-
-											const dir = path.dirname(to);
-											fs.existsSync(dir) || wrench.mkdirSyncRecursive(dir);
-
-											const exists = fs.existsSync(to);
-											// dest doesn't exist, or new contents differs from existing dest file
-											if (!exists || newContents !== fs.readFileSync(to).toString()) {
-												this.logger.debug(__('Copying and minifying %s => %s', from.cyan, to.cyan));
-												// no need to delete if it exists, writeFile will overwrite anyways
-												fs.writeFileSync(to, newContents);
-												this.jsFilesChanged = true;
-											} else {
-												this.logger.trace(__('No change, skipping %s', to.cyan));
-											}
-											cb2();
-										} catch (err) {
-											err.message.split('\n').forEach(this.logger.error);
-											if (err.codeFrame) { // if we have a nicely formatted pointer to syntax error from babel, use it!
-												this.logger.log(err.codeFrame);
-											}
-											this.logger.log();
-											process.exit(1);
-										} finally {
-											this.unmarkBuildDirFile(to);
-										}
-									})(r, from, to, cb);
-								})(info.src, info.dest, next);
-							} catch (ex) {
-								ex.message.split('\n').forEach(this.logger.error);
-								this.logger.log();
-								process.exit(1);
-							}
-						}.bind(this));
-					}.bind(this), next);
+					const task = new ProcessJsTask({
+						inputFiles: Object.keys(jsFiles).map(relPath => jsFiles[relPath].src),
+						incrementalDirectory: path.join(this.buildDir, 'incremental', 'process-js'),
+						jsFiles,
+						jsBootstrapFiles,
+						logger: this.logger,
+						builder: this,
+						sdkCommonFolder
+					});
+					task.run().then(next, next);
```

Comment:
<!--
  0 failure: 
  2 warning:  iphone/cli/comman..., iphone/cli/comman...
  
  
  DangerID: danger-id-default;
  File: iphone/cli/commands/_build.js;
  Line: 5830;
-->

- :warning: iphone/cli/commands/_build.js line 5830 – Avoid calling back inside of a promise. (promise/no-callback-in-promise)
- :warning: iphone/cli/commands/_build.js line 5830 – Expected catch() or return (promise/catch-or-return)